File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.java
class Solution {
    public int findSpecialInteger(int[] arr) {
    HashMap<Integer,Integer> hs=new HashMap<>();
    for(int x:arr)
     hs.put(x,hs.getOrDefault(x,0)+1);
    int max=0;
    int res=0;
     for(int i=0;i<arr.length;i++){
        if(hs.get(arr[i])>max){
            max=hs.get(arr[i]);
            res=arr[i];
        }
     }
   return res;
    }
}

File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.js
var findSpecialInteger = function(arr) {
    let hs =new Map();

    let res =arr.length / 4;

    for(let i=0;i<arr.length;i++){
        hs.set(arr[i],(hs.get(arr[i]) || 0)+1);

        if(hs.get(arr[i])>res)
            return arr[i];
    }

    return -1;
};